lib/elixir/lib/inspect/algebra.ex

```diff
@@ -88,8 +93,33 @@ defmodule Inspect.Opts do
           pretty: boolean,
           safe: boolean,
           syntax_colors: [{color_key, IO.ANSI.ansidata()}],
-          inspect_fun: (any, t -> Inspect.Algebra.t())
+          inspect_fun: (any, t -> Inspect.Algebra.t()),
+          custom_options: [{atom, any()}]
         }
+
+  def custom_options(opts) do
+    Enum.reject(opts, fn
```

Comment:
You can replace this `Enum.reject/2` call with

```elixir
Keyword.drop(opts, [:structs, ...])
```

Reply:
also Keyword.drop/2 is desirable because the function will raise if opts is not a keyword list, and with Enum.reject it will pass as is